lib/storage/src/content_manager/consensus_state.rs

```diff
@@ -638,7 +638,8 @@ impl Persistent {
     }
 
     pub fn is_new(&self) -> bool {
-        self.new
+        // State is also new if the peer created it but failed to bootstrap - therefore leaving `term` at 0
+        self.new || self.state.hard_state.term == 0
```

Comment:
Is it possible to enrich the semantic of the `new` field so that it contains this case?
That is not switching `new` to false if the term == 0?

Otherwise maybe rename `new` to something more precise?

Reply:
I remembered another idea that we had - to remove `new` from the state at all. So I tried it, and this way it seems more readable. At least to me :smile:

Reply:
Thank you :+1: